pkg/consuldp/metrics.go

```diff
@@ -266,6 +266,7 @@ func (m *metricsConfig) getPromDefaults() (*prom.Registry, *prometheus.Prometheu
 		return nil, nil, err
 	}
 	opts := &prometheus.PrometheusOpts{
+		Expiration:       m.cfg.Prometheus.RetentionTime,
```

Comment:
This looks like the right thing based on what Consul does: https://github.com/hashicorp/consul/blob/b9f70c0bf1a6db535663158ac943542900c3c757/agent/config/builder.go#L929